components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -548,6 +555,38 @@ export function ConversationContextProvider(props: React.PropsWithChildren) {
     }
   }
 
+  const ignoreExternalLinkWarning = JSON.parse(
+    localStorage.getItem(IGNORE_EXTERNAL_LINK_WARNING_KEY) ?? 'false'
+  )
```

Comment:
I missed this before - I'm not sure we should be reloading and re-parsing this on every render? Should we just do it once via `React.useMemo`, store in a `React.ref` and then update when `setIgnoreExternalLinkWarning` is called? If we want to also update this when another Tab or Sidebar instance of Leo changes this setting then we can add an event handler for `window.addEventListener("storage",`

Reply:
Good idea, moved this to a `ref` and added an `eventListener`!